### PR TITLE
feat(StreamEast): add presence

### PR DESCRIPTION
**Pull request by Dlunhappy226**
## Description 

This PR adds presence for the popular sports streaming platform, StreamEast.

## Acknowledgements
- [x] I read the [Presence Guidelines](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md)
- [x] I linted the code by running `yarn format`
- [x] The PR title follows the repo's [commit conventions](https://github.com/PreMiD/Presences/blob/main/.github/COMMIT_CONVENTION.md)

## Screenshots
<details>
<summary> Proof showing the creation/modification is working as expected </summary>

image

image
</details>

### CODE REVIEW

File: websites/S/StreamEast/metadata.json [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+		"id": "697250122181902436",
+		"name": "dlunhappy226"
+	},
+	"service": "StreamEast",
+	"description": {
+		"en": "StreamEast is a free HD sports streaming platform."
+	},
+	"url": "the.streameast.app",
+	"version": "1.0.0",
+	"logo": "https://i.ibb.co/nMsJbMX/favicon-48x48.png",
```
Comment (theusaf) [inline websites/S/StreamEast/metadata.json:14]: Logo should be 512x512px

**Review by theusaf — CHANGES_REQUESTED** [review]
Also this presence is low-quality. See if there are any additional details that can be added

**Review by Bas950 — CHANGES_REQUESTED** [review]
Too low quality

**Comment by Dlunhappy226** [comment]
The detail is changed based on what the user is watching.
image
image

**Review by theusaf — CHANGES_REQUESTED** [review]
Try adding some of the other pages other than the streams (blog, categories, search, etc.)

File: websites/S/StreamEast/presence.ts [hunk 2]
```diff
@@ -0,0 +1,68 @@
+const presence = new Presence({
+		clientId: "1316661139672797236",
+	}),
+	browsingTimestamp = Math.floor(Date.now() / 1000);
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+		largeImageKey: "https://i.ibb.co/gr5gcV8/file.jpg",
+		startTimestamp: browsingTimestamp,
+	};
```
Comment (darkvillager2) [inline websites/S/StreamEast/presence.ts:10]:
```suggestion
	},
	{pathname} = document.location;
```

File: websites/S/StreamEast/presence.ts [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+	switch (document.location.pathname) {
```
Comment (darkvillager2) [inline websites/S/StreamEast/presence.ts:12]:
```suggestion
	switch (pathname) {
```

File: websites/S/StreamEast/presence.ts [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+		case "/f1/streams": {
+			presenceData.details = "Viewing F1 streams";
+			break;
+		}
+		case "/multi-stream": {
+			presenceData.details = "Watching multiple streams";
+			break;
+		}
+		default:
+			if (document.location.pathname.includes("/blog/"))
```
Comment (darkvillager2) [inline websites/S/StreamEast/presence.ts:54]:
```suggestion
			if (pathname.includes("/blog/"))
```

File: websites/S/StreamEast/presence.ts [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+				presenceData.details = `Viewing blog: ${document.title}`;
+			else if (document.title.includes("vs")) {
+				presenceData.details = `Watching: [${window.location.pathname
```
Comment (darkvillager2) [inline websites/S/StreamEast/presence.ts:57]:
```suggestion
				presenceData.details = `Watching: [${pathname
```

**Review by darkvillager2 — CHANGES_REQUESTED** [review]
Hello,

Thanks for opening this Pull Request. 

I have requested changes to the code.

If you have any questions feel free to ask them here or on discord (@Dark_Ville)

**Review by theusaf — APPROVED** [review]

**Review by Bas950 — APPROVED** [review]

**Pull request merged**